### PR TITLE
Feat: Refactor Models and Compliance Models Added - 151

### PR DESCRIPTION
**Pull request by AlexZorkin**
- Refactored models directory to use context related sub-directories
- Added Compliance models, migration, and seeders

### CODE REVIEW

File: backend/lcfs/db/models/fuel/CustomFuelType.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    __table_args__ = (
+        {'comment': "Lookup table for custom fuel types."}
+    )
+    
+    custom_fuel_type_id = Column(Integer, primary_key=True, autoincrement=True, comment="Unique identifier for the custom fuel type")
+    type = Column(String, nullable=False, comment="Type of the custom fuel")
+    description = Column(String, nullable=True, comment="Description of the custom fuel type")
+
+    fuel_supplies = relationship('FuelSupply', back_populates='custom_fuel_type')
+    other_uses = relationship('OtherUses', back_populates='custom_fuel_type')
```
Comment (prv-proton) [inline backend/lcfs/db/models/fuel/CustomFuelType.py:16]: other_uses  relationship is not defined in OtherUses table.

File: backend/lcfs/db/models/compliance/ComplianceReport.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+from sqlalchemy.orm import relationship
+from lcfs.db.base import BaseModel, Auditable
+
+class ComplianceReport(BaseModel, Auditable):
+    __tablename__ = 'compliance_report'
+    __table_args__ = (
+        {'comment': "Main tracking table for all the sub-tables associated with a supplier's annual compliance report"}
+    )
+    
+    compliance_report_id = Column(Integer, primary_key=True, autoincrement=True, comment="Unique identifier for the compliance report")
```
Comment (prv-proton) [inline backend/lcfs/db/models/compliance/ComplianceReport.py:11]: There is a type column in the compliance report index table in the UI, which has columns like "Quarter 1 report, Compliance Report, Supplementary report", do we need to take this into consideration?

File: backend/lcfs/db/models/compliance/ComplianceReportHistory.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+class ComplianceReportHistory(BaseModel, Auditable):
+    __tablename__ = 'compliance_report_history'
+    __table_args__ = (
+        {'comment': "Tracks status changes of compliance reports"}
+    )
+    
+    compliance_report_history_id = Column(Integer, primary_key=True, autoincrement=True, comment="Unique identifier for the compliance report history")
+    compliance_report_id = Column(Integer, ForeignKey('compliance_report.compliance_report_id'), nullable=False, comment="Foreign key to the compliance report")
+    status_id = Column(Integer, ForeignKey('compliance_report_status.compliance_report_status_id'), nullable=False, comment="Foreign key to the compliance report status")
+    user_role_id = Column(Integer, nullable=False, comment="Identifier for the user role associated with the status change")
```
Comment (prv-proton) [inline backend/lcfs/db/models/compliance/ComplianceReportHistory.py:14]: Shouldn't this be user_profile_id, similar to transfer history?

File: backend/lcfs/db/models/compliance/ComplianceReport.py (hunk 2 above)
Comment (AlexZorkin) [reply backend/lcfs/db/models/compliance/ComplianceReport.py:11]: We're not going to tackle supplemental and quarterly reporting later. I think its better to just focus on the static page for now.

File: backend/lcfs/db/models/compliance/ComplianceReportHistory.py (hunk 3 above)
Comment (AlexZorkin) [reply backend/lcfs/db/models/compliance/ComplianceReportHistory.py:14]: this was a mistake, should've been user_profile_id.

**Pull request merged**